github/resource_github_membership_test.go

```diff
@@ -140,3 +162,18 @@ var testAccGithubMembershipConfig string = fmt.Sprintf(`
     role = "member"
   }
 `, testCollaborator)
+
+func testAccGithubMembershipConfig_caseInsensitive() string {
+	otherCase := []rune(testCollaborator)
```

Comment:
If I get the intent right, you are modifying the collaborator username, by toggling the case of the first rune, to test case insensitivity functionality. Is that correct? If so what about changing the function signature to take in an input, and just pass the collaborator name with the expected case. 

```go
func testAccGithubMembershipConfig_caseInsensitive(memberName string) {
return fmt.Sprintf(`
  resource "github_membership" "test_org_membership" {
    username = "%s"
    role = "member"
  }
`, memberName)
}
```

This way in the acceptance test you can run two TestSteps:

```go
Steps: []resource.TestStep{
			{
				Config: testAccGithubMembershipConfig_caseInsensitive(strings.ToUpper(testCollaborator)),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckGithubMembershipExists("github_membership.test_org_membership", &membership),
				),
			},
			{
				Config: testAccGithubMembershipConfig_caseInsensitive(strings.ToLower(testCollaborator)),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckGithubMembershipExists("github_membership.test_org_membership", &membership),
				),
			},
		},
```